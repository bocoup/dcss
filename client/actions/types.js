export const LOG_IN = 'LOG_IN';
export const LOG_OUT = 'LOG_OUT';

export const GET_CATEGORIES_SUCCESS = 'GET_CATEGORIES_SUCCESS';
export const GET_CATEGORIES_ERROR = 'GET_CATEGORIES_ERROR';

export const SET_USERS = 'SET_USERS';

export const GET_USER = 'GET_USER';
export const GET_USER_SUCCESS = 'GET_USER_SUCCESS';
export const GET_USER_ERROR = 'GET_USER_ERROR';

export const GET_SESSION_SUCCESS = 'GET_SESSION_SUCCESS';
export const GET_SESSION_ERROR = 'GET_SESSION_ERROR';

export const SET_USER = 'SET_USER';
export const SET_USER_SUCCESS = 'SET_USER_SUCCESS';
export const SET_USER_ERROR = 'SET_USER_ERROR';

export const GET_USERS = 'GET_USERS';
export const GET_USERS_SUCCESS = 'GET_USERS_SUCCESS';
export const GET_USERS_ERROR = 'GET_USERS_ERROR';

export const GET_RESPONSE = 'GET_RESPONSE';
export const GET_RESPONSE_SUCCESS = 'GET_RESPONSE_SUCCESS';
export const GET_RESPONSE_ERROR = 'GET_RESPONSE_ERROR';

export const GET_TRANSCRIPT = 'GET_TRANSCRIPT';
export const GET_TRANSCRIPT_SUCCESS = 'GET_TRANSCRIPT_SUCCESS';
export const GET_TRANSCRIPT_ERROR = 'GET_TRANSCRIPT_ERROR';

export const GET_RESPONSES = 'GET_RESPONSES';
export const GET_RESPONSES_SUCCESS = 'GET_RESPONSES_SUCCESS';
export const GET_RESPONSES_ERROR = 'GET_RESPONSE_ERROR';

export const SET_RESPONSE = 'SET_RESPONSE';
export const SET_RESPONSE_SUCCESS = 'SET_RESPONSE_SUCCESS';
export const SET_RESPONSE_ERROR = 'SET_RESPONSE_ERROR';

export const SET_RESPONSES = 'SET_RESPONSES';
export const SET_RESPONSES_SUCCESS = 'SET_RESPONSES_SUCCESS';
export const SET_RESPONSES_ERROR = 'SET_RESPONSES_ERROR';

export const GET_RUN = 'GET_RUN';
export const GET_RUN_SUCCESS = 'GET_RUN_SUCCESS';
export const GET_RUN_ERROR = 'GET_RUN_ERROR';

export const GET_RUNS = 'GET_RUNS';
export const GET_RUNS_SUCCESS = 'GET_RUNS_SUCCESS';
export const GET_RUNS_ERROR = 'GET_RUNS_ERROR';

export const GET_RUN_DATA = 'GET_RUN_DATA';
export const GET_RUN_DATA_SUCCESS = 'GET_RUN_DATA_SUCCESS';
export const GET_RUN_DATA_ERROR = 'GET_RUN_DATA_ERROR';

export const SET_USER_ROLE_SUCCESS = 'SET_USER_ROLE_SUCCESS';
export const SET_USER_ROLE_ERROR = 'SET_USER_ROLE_ERROR';

export const SET_RUN = 'SET_RUN';
export const SET_RUN_SUCCESS = 'SET_RUN_SUCCESS';
export const SET_RUN_ERROR = 'SET_RUN_ERROR';

export const DELETE_SCENARIO_SUCCESS = 'DELETE_SCENARIO_SUCCESS';
export const DELETE_SCENARIO_ERROR = 'DELETE_SCENARIO_ERROR';

export const UNLOCK_SCENARIO_SUCCESS = 'UNLOCK_SCENARIO_SUCCESS';
export const UNLOCK_SCENARIO_ERROR = 'UNLOCK_SCENARIO_ERROR';

export const GET_SCENARIO = 'GET_SCENARIO';
export const GET_SCENARIO_SUCCESS = 'GET_SCENARIO_SUCCESS';
export const GET_SCENARIO_ERROR = 'GET_SCENARIO_ERROR';

export const GET_SCENARIOS = 'GET_SCENARIOS';
export const GET_SCENARIOS_SUCCESS = 'GET_SCENARIOS_SUCCESS';
export const GET_SCENARIOS_ERROR = 'GET_SCENARIOS_ERROR';

export const SET_SCENARIO = 'SET_SCENARIO';
export const SET_SCENARIO_SUCCESS = 'SET_SCENARIO_SUCCESS';
export const SET_SCENARIO_ERROR = 'SET_SCENARIO_ERROR';

export const GET_RUN_HISTORY = 'GET_RUN_HISTORY';
export const GET_RUN_HISTORY_SUCCESS = 'GET_RUN_HISTORY_SUCCESS';
export const GET_RUN_HISTORY_ERROR = 'GET_RUN_HISTORY_ERROR';

export const SET_SCENARIOS = 'SET_SCENARIOS';
export const SET_SCENARIOS_SUCCESS = 'SET_SCENARIOS_SUCCESS';
export const SET_SCENARIOS_ERROR = 'SET_SCENARIOS_ERROR';

export const DELETE_SLIDE_SUCCESS = 'DELETE_SLIDE_SUCCESS';
export const DELETE_SLIDE_ERROR = 'DELETE_SLIDE_ERROR';

export const SET_SLIDES = 'SET_SLIDES';

export const GET_SLIDES = 'GET_SLIDES';
export const GET_SLIDES_SUCCESS = 'GET_SLIDES_SUCCESS';
export const GET_SLIDES_ERROR = 'GET_SLIDES_ERROR';

export const CREATE_COHORT = 'CREATE_COHORT';
export const CREATE_COHORT_SUCCESS = 'CREATE_COHORT_SUCCESS';
export const CREATE_COHORT_ERROR = 'CREATE_COHORT_ERROR';

export const SET_COHORT = 'SET_COHORT';
export const SET_COHORT_SUCCESS = 'SET_COHORT_SUCCESS';
export const SET_COHORT_ERROR = 'SET_COHORT_ERROR';

export const SET_COHORT_USER_ROLE = 'SET_COHORT_USER_ROLE';
export const SET_COHORT_USER_ROLE_SUCCESS = 'SET_COHORT_USER_ROLE_SUCCESS';
export const SET_COHORT_USER_ROLE_ERROR = 'SET_COHORT_USER_ROLE_ERROR';

export const GET_COHORT = 'GET_COHORT';
export const GET_COHORT_SUCCESS = 'GET_COHORT_SUCCESS';
export const GET_COHORT_ERROR = 'GET_COHORT_ERROR';

export const GET_COHORT_DATA = 'GET_COHORT_DATA';
export const GET_COHORT_RUN_DATA_SUCCESS = 'GET_COHORT_RUN_DATA_SUCCESS';
export const GET_COHORT_RUN_DATA_ERROR = 'GET_COHORT_RUN_DATA_ERROR';

export const GET_COHORT_PARTICIPANTS = 'GET_COHORT_PARTICIPANTS';
export const GET_COHORT_PARTICIPANTS_SUCCESS =
  'GET_COHORT_PARTICIPANTS_SUCCESS';
export const GET_COHORT_PARTICIPANTS_ERROR = 'GET_COHORT_PARTICIPANTS_ERROR';

export const LINK_RUN_TO_COHORT = 'LINK_RUN_TO_COHORT';
export const LINK_RUN_TO_COHORT_SUCCESS = 'LINK_RUN_TO_COHORT_SUCCESS';
export const LINK_RUN_TO_COHORT_ERROR = 'LINK_RUN_TO_COHORT_ERROR';

export const GET_ALL_COHORTS = 'GET_ALL_COHORTS';
export const GET_ALL_COHORTS_SUCCESS = 'GET_ALL_COHORTS_SUCCESS';
export const GET_ALL_COHORTS_ERROR = 'GET_ALL_COHORTS_ERROR';

export const GET_USER_COHORTS = 'GET_USER_COHORTS';
export const GET_USER_COHORTS_SUCCESS = 'GET_USER_COHORTS_SUCCESS';
export const GET_USER_COHORTS_ERROR = 'GET_USER_COHORTS_ERROR';
