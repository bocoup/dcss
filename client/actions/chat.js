import {
  GET_CHAT_ERROR,
  GET_CHAT_SUCCESS,
  GET_CHATS_ERROR,
  GET_CHATS_SUCCESS,
  SET_CHAT_ERROR,
  SET_CHAT_SUCCESS,
  LINK_CHAT_TO_RUN_ERROR,
  LINK_CHAT_TO_RUN_SUCCESS
} from './types';

export let getChat = id => async dispatch => {
  try {
    const res = await (await fetch(`/api/chats/${id}`)).json();

    if (res.error) {
      throw res;
    }
    const { chat } = res;
    dispatch({ type: GET_CHAT_SUCCESS, chat });
    return chat;
  } catch (error) {
    dispatch({ type: GET_CHAT_ERROR, error });
    return null;
  }
};

export let setChat = (id, params) => async dispatch => {
  try {
    if (Object.values(params).length) {
      const body = JSON.stringify(params);
      const res = await (await fetch(`/api/chats/${id}`, {
        method: 'PUT',
        headers: {
          'Content-Type': 'application/json'
        },
        body
      })).json();

      if (res.error) {
        throw res;
      }

      const { chat } = res;
      dispatch({ type: SET_CHAT_SUCCESS, chat });
      return chat;
    }
    return null;
  } catch (error) {
    dispatch({ type: SET_CHAT_ERROR, error });
    return null;
  }
};

export let createChat = params => async dispatch => {
  try {
    if (Object.values(params).length) {
      const { lobby_id } = params;
      const body = JSON.stringify({
        lobby_id
      });
      const res = await (await fetch(`/api/chats`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body
      })).json();

      if (res.error) {
        throw res;
      }
      const { chat } = res;
      dispatch({ type: GET_CHAT_SUCCESS, chat });
      return chat;
    }
    return null;
  } catch (error) {
    dispatch({ type: GET_CHAT_ERROR, error });
    return null;
  }
};

export let getChats = () => async dispatch => {
  try {
    const res = await (await fetch('/api/chats')).json();

    if (res.error) {
      throw res;
    }

    const { chats = [] } = res;
    dispatch({ type: GET_CHATS_SUCCESS, chats });
    return chats;
  } catch (error) {
    dispatch({ type: GET_CHATS_ERROR, error });
    return null;
  }
};

export let linkChatToRun = (chat_id, run_id) => async dispatch => {
  try {
    const res = await (await fetch(
      `/api/chats/link/${chat_id}/run/${run_id}`
    )).json();

    if (res.error) {
      throw res;
    }
    const { chat } = res;
    dispatch({ type: LINK_CHAT_TO_RUN_SUCCESS, chat });
    return chat;
  } catch (error) {
    dispatch({ type: LINK_CHAT_TO_RUN_ERROR, error });
    return null;
  }
};
