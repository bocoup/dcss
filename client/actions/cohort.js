import {
  // CREATE_COHORT,
  CREATE_COHORT_SUCCESS,
  CREATE_COHORT_ERROR,
  SET_COHORT_SUCCESS,
  SET_COHORT_ERROR,
  // GET_COHORT,
  GET_COHORT_SUCCESS,
  GET_COHORT_ERROR,
  // GET_COHORT_PARTICIPANTS,
  GET_COHORT_PARTICIPANTS_SUCCESS,
  GET_COHORT_PARTICIPANTS_ERROR,
  // GET_COHORT_RUN_DATA_SUCCESS,
  GET_COHORT_RUN_DATA_ERROR,
  // GET_ALL_COHORTS,
  GET_ALL_COHORTS_SUCCESS,
  GET_ALL_COHORTS_ERROR,
  // GET_USER_COHORTS,
  GET_USER_COHORTS_SUCCESS,
  GET_USER_COHORTS_ERROR,
  // LINK_COHORT_TO_RUN,
  LINK_RUN_TO_COHORT_SUCCESS,
  LINK_RUN_TO_COHORT_ERROR,
  // SET_COHORT_USER_ROLE,
  SET_COHORT_USER_ROLE_SUCCESS,
  SET_COHORT_USER_ROLE_ERROR
} from './types';

export const createCohort = ({ name }) => async dispatch => {
  try {
    const res = await (await fetch('/api/cohort', {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ name })
    })).json();

    if (res.error) {
      throw res;
    }
    const { cohort } = res;

    dispatch({ type: CREATE_COHORT_SUCCESS, cohort });
    // return the cohort to the promise action for redirection purposes
    return cohort;
  } catch (error) {
    dispatch({ type: CREATE_COHORT_ERROR, error });
    return null;
  }
};

export const setCohort = cohort => async dispatch => {
  // dispatch({ type: SET_COHORT, cohort });
  try {
    const { scenarios } = cohort;
    const body = JSON.stringify({
      scenarios
    });
    await (await fetch(`/api/cohort/${cohort.id}/scenarios`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body
    })).json();
    dispatch({ type: SET_COHORT_SUCCESS, cohort });
  } catch (error) {
    dispatch({ type: SET_COHORT_ERROR, error });
    return null;
  }
};

export const getCohort = id => async dispatch => {
  if (Number.isNaN(id)) {
    return;
  }
  try {
    const res = await (await fetch(`/api/cohort/${id}`)).json();
    if (res.error) {
      throw res;
    }
    const { cohort } = res;
    dispatch({ type: GET_COHORT_SUCCESS, cohort });
    // return the cohort to the promise action for redirection purposes
    return cohort;
  } catch (error) {
    dispatch({ type: GET_COHORT_ERROR, error });
    return null;
  }
};

export const getCohorts = () => async dispatch => {
  try {
    const response = await (await fetch('/api/cohort/my')).json();

    if (response.error) {
      throw response;
    }
    const { cohorts } = response;
    dispatch({ type: GET_USER_COHORTS_SUCCESS, cohorts });
  } catch (error) {
    dispatch({ type: GET_USER_COHORTS_ERROR, error });
    return null;
  }
};

export const getAllCohorts = () => async dispatch => {
  try {
    const res = await (await fetch('/api/cohort/all')).json();
    if (res.error) {
      throw res;
    }
    const { cohorts } = res;
    dispatch({ type: GET_ALL_COHORTS_SUCCESS, cohorts });
    return cohorts;
  } catch (error) {
    dispatch({ type: GET_ALL_COHORTS_ERROR, error });
    return null;
  }
};

export const getCohortParticipants = id => async dispatch => {
  try {
    const res = await (await fetch(`/api/cohort/${id}`)).json();

    if (res.error) {
      throw res;
    }
    const {
      cohort,
      cohort: { users }
    } = res;
    // Dispatch the entire "cohort", but only the "users" property
    // will be used in the reducer.
    dispatch({ type: GET_COHORT_PARTICIPANTS_SUCCESS, cohort });
    // return the cohort to the promise action for redirection purposes
    return users;
  } catch (error) {
    dispatch({ type: GET_COHORT_PARTICIPANTS_ERROR, error });
    return null;
  }
};

export const getCohortData = (
  cohort_id,
  participant_id,
  scenario_id
) => async dispatch => {
  try {
    const endpoint = scenario_id
      ? `/api/cohort/${cohort_id}/scenario/${scenario_id}`
      : `/api/cohort/${cohort_id}/participant/${participant_id}`;

    const res = await (await fetch(endpoint)).json();

    if (res.error) {
      throw res;
    }

    const { prompts, responses } = res;

    // This dispatch does not update internal state, since
    // such state would be thrown away:
    // dispatch({ type: GET_COHORT_RUN_DATA_SUCCESS, prompts, responses });
    //
    return { prompts, responses };
  } catch (error) {
    dispatch({ type: GET_COHORT_RUN_DATA_ERROR, error });
    return null;
  }
};

export const linkRunToCohort = (cohort_id, run_id) => async dispatch => {
  try {
    const res = await (await fetch(
      `/api/cohort/${cohort_id}/run/${run_id}`
    )).json();
    if (res.error) {
      throw res;
    }
    const { cohort } = res;
    dispatch({ type: LINK_RUN_TO_COHORT_SUCCESS });
    dispatch({ type: GET_COHORT_SUCCESS, cohort });
    return cohort;
  } catch (error) {
    dispatch({ type: LINK_RUN_TO_COHORT_ERROR, error });
    return null;
  }
};
// This is used to
export const linkUserToCohort = (cohort_id, role) => async dispatch => {
  try {
    const users = await (await fetch(
      `/api/cohort/${cohort_id}/join/${role}`
    )).json();
    if (users.error) {
      throw users;
    }
    dispatch({ type: SET_COHORT_USER_ROLE_SUCCESS, users });
  } catch (error) {
    dispatch({ type: SET_COHORT_USER_ROLE_ERROR, error });
    return null;
  }
};

export const addCohortUserRole = (
  cohort_id,
  user_id,
  role
) => async dispatch => {
  try {
    const body = JSON.stringify({
      cohort_id,
      user_id,
      roles: [role]
    });
    const result = await (await fetch('/api/cohort/${cohort_id}/roles/add', {
      headers: {
        'Content-Type': 'application/json'
      },
      method: 'POST',
      body
    })).json();

    const { cohort } = result;
    dispatch({ type: SET_COHORT_USER_ROLE_SUCCESS });
    dispatch({ type: GET_COHORT_SUCCESS, cohort });
    return result;
  } catch (error) {
    dispatch({ type: SET_COHORT_USER_ROLE_ERROR, error });
    return null;
  }
};

export const deleteCohortUserRole = (
  cohort_id,
  user_id,
  role
) => async dispatch => {
  try {
    const body = JSON.stringify({
      cohort_id,
      user_id,
      roles: [role]
    });
    const result = await (await fetch('/api/cohort/${cohort_id}/roles/delete', {
      headers: {
        'Content-Type': 'application/json'
      },
      method: 'POST',
      body
    })).json();
    const { cohort } = result;
    dispatch({ type: SET_COHORT_USER_ROLE_SUCCESS });
    dispatch({ type: GET_COHORT_SUCCESS, cohort });
    return result;
  } catch (error) {
    dispatch({ type: SET_COHORT_USER_ROLE_ERROR, error });
    return null;
  }
};
