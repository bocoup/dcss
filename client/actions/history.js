import { GET_RUN_HISTORY_SUCCESS, GET_RUN_HISTORY_ERROR } from './types';

export const getHistoryForScenario = (
  scenario_id,
  cohort_id
) => async dispatch => {
  try {
    const endpoint = cohort_id
      ? `/api/history/${scenario_id}/cohort/${cohort_id}`
      : `/api/history/${scenario_id}`;

    const res = await (await fetch(endpoint)).json();

    if (res.error) {
      throw res;
    }

    const { history } = res;

    dispatch({ type: GET_RUN_HISTORY_SUCCESS, history });
    return { ...history };
  } catch (error) {
    dispatch({ type: GET_RUN_HISTORY_ERROR, error });
    return null;
  }
};
