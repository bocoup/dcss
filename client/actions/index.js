import {
    LOG_IN,
    LOG_OUT,
    SET_RUN,
    SET_SCENARIO,
    SET_SCENARIOS,
    SET_SLIDES,
    SET_USERS
} from './types';

export const logIn = username => ({
    type: LOG_IN,
    payload: {
        username,
        isLoggedIn: true
    }
});

export const logOut = username => ({
    type: LOG_OUT,
    payload: {
        username,
        isLoggedIn: false
    }
});

export const setRun = payload => ({
    type: SET_RUN,
    payload
});

export const setScenario = payload => ({
    type: SET_SCENARIO,
    payload
});

export const setScenarios = payload => ({
    type: SET_SCENARIOS,
    payload
});

export const setSlides = payload => ({
    type: SET_SLIDES,
    payload
});

export const setUsers = payload => ({
    type: SET_USERS,
    payload
});
