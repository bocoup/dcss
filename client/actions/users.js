import { GET_USERS_SUCCESS, GET_USERS_ERROR } from './types';

export const setUsers = users => ({
  type: GET_USERS_SUCCESS,
  users
});

export const getUsers = () => async dispatch => {
  try {
    const { users } = await (await fetch('/api/roles/all')).json();
    dispatch({ type: GET_USERS_SUCCESS, users });
    return users;
  } catch (error) {
    dispatch({ type: GET_USERS_ERROR, error });
    return error;
  }
};
