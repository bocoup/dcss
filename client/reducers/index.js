import { combineReducers } from 'redux';

import login from './login';
import scenario from './scenario';
import admin from './admin';

export default combineReducers({ login, scenario, admin });
