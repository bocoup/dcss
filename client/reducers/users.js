import { GET_USERS_SUCCESS } from '@client/actions/types';

export const users = (state = [], action) => {
  const { users, type } = action;
  switch (type) {
    case GET_USERS_SUCCESS:
      return [...state, ...users];
    default:
      return state;
  }
};
