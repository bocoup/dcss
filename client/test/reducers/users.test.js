import assert from 'assert';
import { state } from '../bootstrap';

import * as reducer from '../../reducers/users';
import * as types from '../../actions/types';

const initialState = [];
const initialStateById = {};

const error = new Error('something unexpected happened on the server');
const original = JSON.parse(JSON.stringify(state));

describe('users', () => {
  let users;
  let usersById;

  beforeEach(() => {
    users = [
      ...original.users
    ];

    usersById = {
      ...original.usersById
    };

  });

  test('initial state', () => {
    expect(reducer.users(undefined, {})).toEqual(initialState);
    expect(reducer.users(undefined, {})).toEqual(initialState);
    expect(reducer.usersById(undefined, {})).toEqual(initialStateById);
    expect(reducer.usersById(undefined, {})).toEqual(initialStateById);
  });

  test('GET_USERS_SUCCESS', () => {
    const action = {
      type: types.GET_USERS_SUCCESS,
      users
    };
    expect(reducer.users(undefined, action)).toMatchSnapshot();
    expect(reducer.users(undefined, action)).toMatchSnapshot();
  });

  test('GET_USERS_SUCCESS dedupe', () => {
    const action = {
      type: types.GET_USERS_SUCCESS,
      users: [...users, ...users, ...users]
    };
    expect(reducer.users(undefined, action)).toMatchSnapshot();
    expect(reducer.users(undefined, action)).toMatchSnapshot();
  });

  test('GET_USERS_SUCCESS', () => {
    const action = {
      type: types.GET_USERS_SUCCESS,
      users
    };
    expect(reducer.usersById(undefined, action)).toMatchSnapshot();
    expect(reducer.usersById(undefined, action)).toMatchSnapshot();
  });


  test('GET_USERS_SUCCESS dedupe', () => {
    const action = {
      type: types.GET_USERS_SUCCESS,
      users: [...users, ...users, ...users]
    };
    expect(reducer.usersById(undefined, action)).toMatchSnapshot();
    expect(reducer.usersById(undefined, action)).toMatchSnapshot();
  });
});
