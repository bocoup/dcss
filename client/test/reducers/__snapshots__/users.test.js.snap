// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users GET_USERS_SUCCESS 1`] = `
Array [
  Object {
    "email": "super@email.com",
    "id": 999,
    "is_anonymous": false,
    "is_super": true,
    "personalname": "Super User",
    "roles": Array [
      "participant",
      "super_admin",
      "facilitator",
      "researcher",
    ],
    "username": "super",
  },
  Object {
    "email": null,
    "id": 4,
    "is_anonymous": true,
    "is_super": false,
    "personalname": null,
    "roles": Array [
      "participant",
    ],
    "username": "credible-lyrebird",
  },
  Object {
    "email": "heartfull-cougar@email.com",
    "id": 3,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Heartfull Cougar",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "heartfull-cougar",
  },
  Object {
    "email": "perfect-oryx@email.com",
    "id": 5,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Perfect Oryx",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "perfect-oryx",
  },
]
`;

exports[`users GET_USERS_SUCCESS 2`] = `
Array [
  Object {
    "email": "super@email.com",
    "id": 999,
    "is_anonymous": false,
    "is_super": true,
    "personalname": "Super User",
    "roles": Array [
      "participant",
      "super_admin",
      "facilitator",
      "researcher",
    ],
    "username": "super",
  },
  Object {
    "email": null,
    "id": 4,
    "is_anonymous": true,
    "is_super": false,
    "personalname": null,
    "roles": Array [
      "participant",
    ],
    "username": "credible-lyrebird",
  },
  Object {
    "email": "heartfull-cougar@email.com",
    "id": 3,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Heartfull Cougar",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "heartfull-cougar",
  },
  Object {
    "email": "perfect-oryx@email.com",
    "id": 5,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Perfect Oryx",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "perfect-oryx",
  },
]
`;

exports[`users GET_USERS_SUCCESS 3`] = `
Object {
  "3": Object {
    "email": "heartfull-cougar@email.com",
    "id": 3,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Heartfull Cougar",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "heartfull-cougar",
  },
  "4": Object {
    "email": null,
    "id": 4,
    "is_anonymous": true,
    "is_super": false,
    "personalname": null,
    "roles": Array [
      "participant",
    ],
    "username": "credible-lyrebird",
  },
  "5": Object {
    "email": "perfect-oryx@email.com",
    "id": 5,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Perfect Oryx",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "perfect-oryx",
  },
  "999": Object {
    "email": "super@email.com",
    "id": 999,
    "is_anonymous": false,
    "is_super": true,
    "personalname": "Super User",
    "roles": Array [
      "participant",
      "super_admin",
      "facilitator",
      "researcher",
    ],
    "username": "super",
  },
}
`;

exports[`users GET_USERS_SUCCESS 4`] = `
Object {
  "3": Object {
    "email": "heartfull-cougar@email.com",
    "id": 3,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Heartfull Cougar",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "heartfull-cougar",
  },
  "4": Object {
    "email": null,
    "id": 4,
    "is_anonymous": true,
    "is_super": false,
    "personalname": null,
    "roles": Array [
      "participant",
    ],
    "username": "credible-lyrebird",
  },
  "5": Object {
    "email": "perfect-oryx@email.com",
    "id": 5,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Perfect Oryx",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "perfect-oryx",
  },
  "999": Object {
    "email": "super@email.com",
    "id": 999,
    "is_anonymous": false,
    "is_super": true,
    "personalname": "Super User",
    "roles": Array [
      "participant",
      "super_admin",
      "facilitator",
      "researcher",
    ],
    "username": "super",
  },
}
`;

exports[`users GET_USERS_SUCCESS dedupe 1`] = `
Array [
  Object {
    "email": "super@email.com",
    "id": 999,
    "is_anonymous": false,
    "is_super": true,
    "personalname": "Super User",
    "roles": Array [
      "participant",
      "super_admin",
      "facilitator",
      "researcher",
    ],
    "username": "super",
  },
  Object {
    "email": null,
    "id": 4,
    "is_anonymous": true,
    "is_super": false,
    "personalname": null,
    "roles": Array [
      "participant",
    ],
    "username": "credible-lyrebird",
  },
  Object {
    "email": "heartfull-cougar@email.com",
    "id": 3,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Heartfull Cougar",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "heartfull-cougar",
  },
  Object {
    "email": "perfect-oryx@email.com",
    "id": 5,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Perfect Oryx",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "perfect-oryx",
  },
]
`;

exports[`users GET_USERS_SUCCESS dedupe 2`] = `
Array [
  Object {
    "email": "super@email.com",
    "id": 999,
    "is_anonymous": false,
    "is_super": true,
    "personalname": "Super User",
    "roles": Array [
      "participant",
      "super_admin",
      "facilitator",
      "researcher",
    ],
    "username": "super",
  },
  Object {
    "email": null,
    "id": 4,
    "is_anonymous": true,
    "is_super": false,
    "personalname": null,
    "roles": Array [
      "participant",
    ],
    "username": "credible-lyrebird",
  },
  Object {
    "email": "heartfull-cougar@email.com",
    "id": 3,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Heartfull Cougar",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "heartfull-cougar",
  },
  Object {
    "email": "perfect-oryx@email.com",
    "id": 5,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Perfect Oryx",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "perfect-oryx",
  },
]
`;

exports[`users GET_USERS_SUCCESS dedupe 3`] = `
Object {
  "3": Object {
    "email": "heartfull-cougar@email.com",
    "id": 3,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Heartfull Cougar",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "heartfull-cougar",
  },
  "4": Object {
    "email": null,
    "id": 4,
    "is_anonymous": true,
    "is_super": false,
    "personalname": null,
    "roles": Array [
      "participant",
    ],
    "username": "credible-lyrebird",
  },
  "5": Object {
    "email": "perfect-oryx@email.com",
    "id": 5,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Perfect Oryx",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "perfect-oryx",
  },
  "999": Object {
    "email": "super@email.com",
    "id": 999,
    "is_anonymous": false,
    "is_super": true,
    "personalname": "Super User",
    "roles": Array [
      "participant",
      "super_admin",
      "facilitator",
      "researcher",
    ],
    "username": "super",
  },
}
`;

exports[`users GET_USERS_SUCCESS dedupe 4`] = `
Object {
  "3": Object {
    "email": "heartfull-cougar@email.com",
    "id": 3,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Heartfull Cougar",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "heartfull-cougar",
  },
  "4": Object {
    "email": null,
    "id": 4,
    "is_anonymous": true,
    "is_super": false,
    "personalname": null,
    "roles": Array [
      "participant",
    ],
    "username": "credible-lyrebird",
  },
  "5": Object {
    "email": "perfect-oryx@email.com",
    "id": 5,
    "is_anonymous": false,
    "is_super": false,
    "personalname": "Perfect Oryx",
    "roles": Array [
      "participant",
      "facilitator",
    ],
    "username": "perfect-oryx",
  },
  "999": Object {
    "email": "super@email.com",
    "id": 999,
    "is_anonymous": false,
    "is_super": true,
    "personalname": "Super User",
    "roles": Array [
      "participant",
      "super_admin",
      "facilitator",
      "researcher",
    ],
    "username": "super",
  },
}
`;
